azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -903,15 +922,20 @@ func resourceArmKubernetesClusterRead(d *schema.ResourceData, meta interface{})
 	if props := resp.ManagedClusterProperties; props != nil {
 		d.Set("dns_prefix", props.DNSPrefix)
 		d.Set("fqdn", props.Fqdn)
+		d.Set("private_fqdn", props.PrivateFQDN)
 		d.Set("kubernetes_version", props.KubernetesVersion)
 		d.Set("node_resource_group", props.NodeResourceGroup)
 		d.Set("enable_pod_security_policy", props.EnablePodSecurityPolicy)
 
-		apiServerAuthorizedIPRanges := utils.FlattenStringSlice(props.APIServerAuthorizedIPRanges)
+		apiServerAuthorizedIPRanges := utils.FlattenStringSlice(props.APIServerAccessProfile.AuthorizedIPRanges)
 		if err := d.Set("api_server_authorized_ip_ranges", apiServerAuthorizedIPRanges); err != nil {
 			return fmt.Errorf("Error setting `api_server_authorized_ip_ranges`: %+v", err)
 		}
 
+		if err := d.Set("private_link_enabled", props.APIServerAccessProfile.EnablePrivateCluster); err != nil {
+			return fmt.Errorf("Error setting `private_link_enabled`: %+v", err)
+		}
```

Comment:
(same here) since it's possible for `props.APIServerAccessProfile` to be nil here (e.g. due to an older or bad API response) - could we nil-check these via:

```suggestion
		if profile := props.APIServerAccessProfile; profile != nil {
			apiServerAuthorizedIPRanges := utils.FlattenStringSlice(profile.AuthorizedIPRanges)
			if err := d.Set("api_server_authorized_ip_ranges", apiServerAuthorizedIPRanges); err != nil {
				return fmt.Errorf("Error setting `api_server_authorized_ip_ranges`: %+v", err)
			}

			d.Set("private_link_enabled", profile.EnablePrivateCluster)
		}
```

also since `private_link_enabled` is a boolean (not a complex type) we can just call d.Set without the error check

Reply:
Resolved in new commit

Reply:
looks like this one was only fixed in the Data Source?